x/upgrade/abci_test.go

```diff
@@ -8,53 +8,70 @@ import (
 	"time"
 
 	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
 	abci "github.com/tendermint/tendermint/abci/types"
 	tmproto "github.com/tendermint/tendermint/proto/tendermint/types"
 	"golang.org/x/exp/maps"
 
-	"cosmossdk.io/depinject"
+	"github.com/cosmos/cosmos-sdk/baseapp"
 	"github.com/cosmos/cosmos-sdk/client/flags"
-	"github.com/cosmos/cosmos-sdk/codec"
 	"github.com/cosmos/cosmos-sdk/server"
+	"github.com/cosmos/cosmos-sdk/testutil"
 	simtestutil "github.com/cosmos/cosmos-sdk/testutil/sims"
 	sdk "github.com/cosmos/cosmos-sdk/types"
 	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
 	"github.com/cosmos/cosmos-sdk/types/module"
-	govtypes "github.com/cosmos/cosmos-sdk/x/gov/types/v1beta1"
+	moduletestutil "github.com/cosmos/cosmos-sdk/types/module/testutil"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	govtypes "github.com/cosmos/cosmos-sdk/x/gov/types"
+	govtypesv1beta1 "github.com/cosmos/cosmos-sdk/x/gov/types/v1beta1"
 	"github.com/cosmos/cosmos-sdk/x/upgrade"
 	"github.com/cosmos/cosmos-sdk/x/upgrade/keeper"
-	"github.com/cosmos/cosmos-sdk/x/upgrade/testutil"
+	"github.com/tendermint/tendermint/libs/log"
+
 	"github.com/cosmos/cosmos-sdk/x/upgrade/types"
 )
 
 type TestSuite struct {
+	suite.Suite
+
 	module  module.BeginBlockAppModule
 	keeper  keeper.Keeper
 	querier sdk.Querier
-	handler govtypes.Handler
+	handler govtypesv1beta1.Handler
 	ctx     sdk.Context
+	baseApp *baseapp.BaseApp
+	encCfg  moduletestutil.TestEncodingConfig
 }
 
 var s TestSuite
 
 func setupTest(t *testing.T, height int64, skip map[int64]bool) TestSuite {
 	var (
-		legacyAmino *codec.LegacyAmino
-		appOptions  = make(simtestutil.AppOptionsMap, 0)
+		appOptions = make(simtestutil.AppOptionsMap, 0)
 	)
 
 	appOptions[server.FlagUnsafeSkipUpgrades] = maps.Keys(skip)
 	appOptions[flags.FlagHome] = t.TempDir()
 
-	appConfig := depinject.Configs(testutil.AppConfig, depinject.Supply(appOptions))
-	app, err := simtestutil.Setup(appConfig, &s.keeper, &legacyAmino)
-	s.keeper.SetVersionSetter(app.BaseApp)
-	require.NoError(t, err)
+	s.encCfg = moduletestutil.MakeTestEncodingConfig(upgrade.AppModuleBasic{})
+	key := sdk.NewKVStoreKey(types.StoreKey)
+	testCtx := testutil.DefaultContextWithDB(s.T(), key, sdk.NewTransientStoreKey("transient_test"))
+
+	s.baseApp = baseapp.NewBaseApp(
+		"upgrade",
+		log.NewNopLogger(),
+		testCtx.DB,
+		s.encCfg.TxConfig.TxDecoder(),
+	)
+
+	s.keeper = keeper.NewKeeper(skip, key, s.encCfg.Codec, "", nil, authtypes.NewModuleAddress(govtypes.ModuleName).String())
```

Comment:
Can we use `t.TempDir` for the homePath? I am uncertain if it is used, but it makes more sense than nothing.

```suggestion
	s.keeper = keeper.NewKeeper(skip, key, s.encCfg.Codec, t.TempDir(), nil, authtypes.NewModuleAddress(govtypes.ModuleName).String())
```